app/src/main/java/de/grobox/transportr/trips/search/DirectionsActivity.java

```diff
@@ -147,6 +152,26 @@ private void processIntent() {
 					to = viewModel.getWork().getValue();
 					if (to == null) new WorkPickerFragment().show(getSupportFragmentManager(), WorkPickerFragment.TAG);
 					break;
+				case INTENT_URI_FAVORITE:
+					from = null;
+					via = null;
+					to = null;
+					long uid = intent.getLongExtra("uid",0);
+
+					LiveData<List<FavoriteTripItem>> tripsLiveData;
+					tripsLiveData = viewModel.getFavoriteTrips();
+					tripsLiveData.observe(this, trips -> {
+						for (FavoriteTripItem trip : trips) {
+							if (trip!=null && trip.getUid()==uid)
+							{
+								searchFromTo(trip.getFrom(), trip.getVia(), trip.getTo());
+								tripsLiveData.removeObservers(this); // everything is done
+								break;
+							}
+						}
+					});
+					setIntent(null);
+					return; // will be finished in viewmodel's loaded trips
```

Comment:
This looks a bit strange. Why don't you let this method go down to the end and assign `from`, `via` and `to` properly instead of suddenly returning here and duplicating code before?

Reply:
The main problem is that we don't know when all trips will be loaded and if they even contain the one we are looking for. So if we just use `getValue()` (as for Home or Work) it might not be loaded in time and we miss it. But we can't block here and wait for it either, since it might never appear.
The only solution I saw, was to attach `searchFromTo` to the observer and exit the method. But I don't have a valid `from` and `to` at this point, so I can't let the method go down to the end.

If you have a better idea, I would be happy to hear it.

I'll fix the other issues you pointed out.

Reply:
Oh, and thank you for taking the time to review this!

Reply:
The main problem is that we don't know when all trips will be loaded and
if they even contain the one we are looking for. So if we just use
`getValue()` (as for Home or Work) it might not be loaded in time and we
miss it. But we can't block here and wait for it either, since it might
never appear.
The only solution I saw, was to attach `searchFromTo` to the observer
and exit the method. But I don't have a valid `from` and `to` at this
point, so I can't let the method go down to the end.

If you have a better idea, I would be happy to hear it.

Thank you for taking the time to review my changes.
I'll fix the other
issues you pointed out.

Reply:
Ah I see. Unfortunately, I don't see a better way to handle this myself.